modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/upgrade/v3_0_1/SAPEntryUpgradeProcess.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.security.service.access.policy.internal.upgrade.v3_0_1;
+
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.service.CompanyLocalServiceUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.security.service.access.policy.service.SAPEntryLocalService;
+
+/**
+ * @author Christopher Kian
+ */
+public class SAPEntryUpgradeProcess extends UpgradeProcess {
+
+	public SAPEntryUpgradeProcess(SAPEntryLocalService sapEntryLocalService) {
+		_sapEntryLocalService = sapEntryLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		CompanyLocalServiceUtil.forEachCompanyId(
+			companyId -> {
+				try {
+					_sapEntryLocalService.checkSystemSAPEntries(companyId);
```

Comment:
@ChrisKian we cannot call the API of the same module because the service-access-policy-service module might not start until its schema is resolved in OSGi.

Reply:
I've made the changes and verified the upgrade process still works as intended.  See this commit https://github.com/liferay-appsec/liferay-portal/pull/1815/commits/be211c4c015dc438d517162b1ba1a896c539b72c